src/metaswitch/sasclient/sender.py

```diff
@@ -118,6 +119,23 @@ def connect(self):
                 self._reconnect_wait = MIN_RECONNECT_WAIT_TIME
                 self._connected = True
 
+        if self._discarding:
+            # Some logs have been discarded prior to this connection attempt.
+            # Make a log to indicate that this has happened, and what is likely
+            # to happen next.
+            msg = ("The SAS client library has filled its message queue, and has discarded "
+                   "some messages.  ")
+
+            if self._connected:
+                msg += ("The connection to SAS is currently active, so the backlog should "
+                        "now clear, allowing new SAS logs to be sent succesfully.")
+                self._discarding = False
+            else:
+                msg += ("There is currently no connection to SAS, so all new SAS logs are "
+                        "currently being discarded.  This will continue until the "
```

Comment:
Really nitpicky but you say 'currently' twice in this error message.

Reply:
I'll fix